unstable test: TestGlobalMemoryTuner

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
```shell
=== RUN   TestGlobalMemoryTuner
    memory_limit_tuner_test.go:82: 
        	Error Trace:	/home/jenkins/.tidb/tmp/5b8829669b2a941b840f9eb369e2b54f/sandbox/processwrapper-sandbox/4009/execroot/__main__/bazel-out/k8-fastbuild/bin/util/gctuner/gctuner_test_/gctuner_test.runfiles/__main__/util/gctuner/memory_limit_tuner_test.go:82
        	            				/home/jenkins/.tidb/tmp/5b8829669b2a941b840f9eb369e2b54f/sandbox/processwrapper-sandbox/4009/execroot/__main__/bazel-out/k8-fastbuild/bin/util/gctuner/gctuner_test_/gctuner_test.runfiles/__main__/util/gctuner/memory_limit_tuner_test.go:108
        	Error:      	Should be true
        	Test:       	TestGlobalMemoryTuner
--- FAIL: TestGlobalMemoryTuner (2.47s)
```
link https://ci.pingcap.net/blue/organizations/jenkins/tidb-unit-test-hourly/detail/tidb-unit-test-hourly/9286/tests
### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/util/gctuner/memory_limit_tuner_test.go
+++ b/util/gctuner/memory_limit_tuner_test.go
@@ -61,13 +61,16 @@ func TestGlobalMemoryTuner(t *testing.T) {
 	defer func() {
 		// If test.count > 1, wait tuning finished.
 		require.Eventually(t, func() bool {
+			//nolint: all_revive
 			return GlobalMemoryLimitTuner.isTuning.Load()
 		}, 5*time.Second, 100*time.Millisecond)
 		require.Eventually(t, func() bool {
-			return GlobalMemoryLimitTuner.waitingReset.Load() == false
+			//nolint: all_revive
+			return !GlobalMemoryLimitTuner.waitingReset.Load()
 		}, 5*time.Second, 100*time.Millisecond)
 		require.Eventually(t, func() bool {
-			return GlobalMemoryLimitTuner.nextGCTriggeredByMemoryLimit.Load() == false
+			//nolint: all_revive
+			return !GlobalMemoryLimitTuner.nextGCTriggeredByMemoryLimit.Load()
 		}, 5*time.Second, 100*time.Millisecond)
 	}()
 

--- a/util/gctuner/memory_limit_tuner_test.go
+++ b/util/gctuner/memory_limit_tuner_test.go
@@ -59,10 +59,16 @@ func TestGlobalMemoryTuner(t *testing.T) {
 	GlobalMemoryLimitTuner.UpdateMemoryLimit()
 	require.True(t, GlobalMemoryLimitTuner.isTuning.Load())
 	defer func() {
-		time.Sleep(1 * time.Second) // If test.count > 1, wait tuning finished.
-		require.True(t, GlobalMemoryLimitTuner.isTuning.Load())
-		require.False(t, GlobalMemoryLimitTuner.waitingReset.Load())
-		require.Equal(t, GlobalMemoryLimitTuner.nextGCTriggeredByMemoryLimit.Load(), false)
+		// If test.count > 1, wait tuning finished.
+		require.Eventually(t, func() bool {
+			return GlobalMemoryLimitTuner.isTuning.Load()
+		}, 5*time.Second, 100*time.Millisecond)
+		require.Eventually(t, func() bool {
+			return GlobalMemoryLimitTuner.waitingReset.Load() == false
+		}, 5*time.Second, 100*time.Millisecond)
+		require.Eventually(t, func() bool {
+			return GlobalMemoryLimitTuner.nextGCTriggeredByMemoryLimit.Load() == false
+		}, 5*time.Second, 100*time.Millisecond)
 	}()
 
 	allocator := &mockAllocator{}

